pest/src/parser_state.rs

```diff
@@ -885,12 +885,19 @@ impl<'i, R: RuleType> ParserState<'i, R> {
     /// assert_eq!(result.unwrap().position().pos(), 4);
     /// ```
     #[inline]
-    pub fn stack_match_peek(self: Box<Self>) -> ParseResult<Box<Self>> {
-        let string = self.stack.iter()
-            .map(|s| s.as_str())
-            .collect::<Vec<&str>>()
-            .concat();
-        self.match_string(&string)
+    pub fn stack_match_peek(mut self: Box<Self>) -> ParseResult<Box<Self>> {
+        let mut position = self.position.clone();
+        let result = self.stack.iter().all(|span| {
+            position.match_string(span.as_str())
+        });
+
+        self.position = position;
```

Comment:
I believe this needs some kind of mechanism to roll-back to the initial position if all the matches weren't successfull. This line should be in the if clause. Same with pop all.

Reply:
Good point. I've made the change 👍